tap-version-14-specification.md

```diff
@@ -0,0 +1,807 @@
+provided _must_ be within the range described by the Plan.
+
+This is valid TAP and a successful test run:
+
+```tap
+TAP version 14
+1..3
+ok 2
+ok 3
+ok 1
+```
+
+This is not a successful test run. Even though there are 3 Test Points,
+the Test Point ID 4 is outside the stated Plan range.
+
+```tap
+TAP version 14
+1..3
+ok 2
+ok 4
+ok 1
+```
+
+Test Point IDs _should_ be unique within the TAP Document.  Harnesses _may_
+warn about repeated Test Point IDs or treat them as a test failure, but
+_must not_ treat a Test Point with a re-used ID as a non-TAP line.
+
+#### Description
+
+Any text after the test number but before a `#` is the description of
+the test point.
+
+```tap
+ok 42 - this is the description of the test
+```
+
+Descriptions _should_ be separated from the Test Point Status and Test
+Point ID by the string `" - "`, in order to prevent confusing a numeric
+description with a Test Point ID.  Harnesses _must_ treat Test Points
+identically whether the description starts with `" - "` or not.
+
+For example, these two test points _must_ be treated identically by a
+Harness:
+
+```tap
+ok 1 this is fine
+ok 1 - this is fine
+```
+
+Harnesses _should not_ consider a leading `" - "` to be a part of the
+description reported to a user.
+
+#### Directive
+
+Directives are special notes that follow the first unescaped `#` on the
+Test Point line.  Only two are currently defined: `TODO` and `SKIP`.
+
+Directives are not case sensitive.  That is, Harnesses _must_ treat `#
+SKIP`, `# skip`, and `# SkIp` identically.
+
+Harnesses _may_ support additional platform-specific Directives.  Future
+versions of this specification may codify additional Directives with
+defined semantics.
+
+Unrecognized Directives _must_ not be treated as test failure, or an
+invalid TAP line.  Harnesses _should_ include any unrecognized directives
+in the Test Point description.
+
+Note that escaped `#` characters are not to be treated as delimiters for
+Directives.  See "Escaping" below.
+
+##### `TODO` tests
+
+If the directive starts with `# TODO`, the test is counted as a todo test,
+and the text after `TODO` is the explanation.
+
+```tap
+not ok 14 # TODO bend space and time
+```
+
+If the TODO has an explanation, it must be separated from TODO by a space.
+These tests represent a feature to be implemented or a bug to be fixed and
+act as something of an executable "things to do" list. They are not
+expected to succeed.
+
+Should a todo test point begin succeeding, the harness _may_ report it in
+some way that indicates that whatever was supposed to be done has been, and
+it should be promoted to a normal Test Point.
+
+Harnesses _must not_ treat failing `TODO` test points as a test failure.
+
+Harneses _should_ report `TODO` test points found as a list of items
+needing work, if that is appropriate for their use case.
+
+##### `SKIP` tests
+
+If the directive starts with `# SKIP`, the test is counted as a skipped
```

Comment:
To be backwards compatible with TAP13 and support `# Skipped`, I think we need the directive to support the regex that permits non-whitespace characters immediately next to SKIP, which is defined in that version of the spec.

> If the directive starts with # SKIP, the test is counted as having been skipped. If the whole test file succeeds, the count of skipped tests is included in the generated output. The harness should report the text after # SKIP\S*\s+ as a reason for skipping.

Reply:
Ahhhh yes. Node-tap doesn't support this today, but I see it should, according to tap13 spec. This is why clear specs are useful! I'll update and fix node-tap.